src/EnergyPlus/OutdoorAirUnit.cc

```diff
@@ -1508,24 +1523,18 @@ namespace OutdoorAirUnit {
 				Node( InletNode ).MassFlowRate = OutAirUnit( OAUnitNum ).ExtAirMassFlow;
 			}
 
-			//Air mass balance check (removed because exhaust and supply can be imbalanced
-			//    IF ((Node(InletNode)%MassFlowRate > Node(OutsideAirNode)%MassFlowRate) &
-			//             .OR.(Node(InletNode)%MassFlowRate < Node(OutsideAirNode)%MassFlowRate)) THEN
-			//      OutAirUnit(OAUnitNum)%UnBalancedErrCount = OutAirUnit(OAUnitNum)%UnBalancedErrCount + 1
-			//      IF (OutAirUnit(OAUnitNum)%UnBalancedErrCount .EQ. 1) THEN
-			//        CALL ShowWarningError('Air mass flow between zone supply and exhaust is not balanced')
-			//        CALL ShowContinueError('Occurs in ' // 'ZoneHVAC:OutdoorAirUnit' // ' Object=' &
-			//                               //TRIM(OutAirUnit(OAUnitNum)%Name))
-			//        CALL ShowContinueError('Air mass balance is required by other outdoor air units,'// &
-			//                                  'ZoneMixing, ZoneCrossMixing, or other air flow control inputs.')
-			//!
-			//!  CALL ShowContinueErrorTimeStamp('Air volume flow rate ratio = '//TRIM(RoundSigDigits(HXAirVolFlowRatio,3))//'.')
-			//!ELSE
-			//! CALL ShowRecurringWarningErrorAtEnd(TRIM(OutAirUnit(OAUnitNum)%Name)//&
-			//! ':  Air mass balance is required by other outdoor air units, ZoneMixing, ZoneCrossMixing, or other air flow control inputs.'&
-			//!   , OutAirUnit(OAUnitNum)%UnBalancedErrIndex)
-			//      END IF
-			//    END IF
+			// Air mass balance check
+			if ( ( std::abs( OutAirUnit( OAUnitNum ).ExtAirMassFlow - OutAirUnit( OAUnitNum ).OutAirMassFlow ) > 0.001 )  && ( !DataHeatBalance::ZoneAirMassFlow.EnforceZoneMassBalance ) ) {
+				OutAirUnit( OAUnitNum ).UnBalancedErrCount = OutAirUnit( OAUnitNum ).UnBalancedErrCount + 1;
+				if ( OutAirUnit( OAUnitNum ).UnBalancedErrCount == 1 ) {
+					ShowWarningError( "Air mass flow between zone supply and exhaust is not balanced" );
+					ShowContinueError( "Occurs in ZoneHVAC:OutdoorAirUnit Object= " + OutAirUnit( OAUnitNum ).Name );
+					ShowContinueError( "Air mass balance is required by other outdoor air units: Fan:ZoneExhaust, ZoneMixing, ZoneCrossMixing, or other air flow control inputs." );
+					ShowContinueErrorTimeStamp( "The outdoor mass flow rate = " + General::RoundSigDigits( OutAirUnit( OAUnitNum ).OutAirMassFlow, 3 ) + " and the exhaust mass flow rate = " + General::RoundSigDigits( OutAirUnit( OAUnitNum ).ExtAirMassFlow, 3 ) + ".");
+				} else {
+					ShowRecurringWarningErrorAtEnd( OutAirUnit( OAUnitNum ).Name + ":  Air mass balance is required by other outdoor air units, Fan:ZoneExhaust, ZoneMixing, ZoneCrossMixing, or other air flow control inputs.", OutAirUnit( OAUnitNum ).UnBalancedErrIndex );
```

Comment:
@lgu1234 @rraustad Since we don't know for sure here if other components are providing the flows to balance the zone, I think we should drop the recurring warning and state in the initial warning that this will only be reported once.  That's what the unbalanced airloop and zone warning do as well in CalcZoneMassBalance.

Reply:
@mjwitte I am going to drop the recurring warnings.